### PR TITLE
ui: changes in migrate vm storage and migrate volume form

### PR DESCRIPTION
**Pull request by shwstppr**
### Description

Better forms in UI for migrating VMs and volumes.

- Show option to migrate with storage while live migrating a VM
- For VM storage migration (stopped VM), allow migrating volumes to specific primary storages
- Show primary storage details in migrate volume form

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### Screenshots (if appropriate):

Migrate VM form
Screenshot from 2021-10-29 12-04-09

Migrate VM storage form
Screenshot from 2021-10-29 12-04-32

Screenshot from 2021-10-29 12-05-38

Migrate volume form
Screenshot from 2021-11-01 15-48-28

### CODE REVIEW

**Comment by sureshanaparti** [comment]
@blueorangutan ui

**Comment by shwstppr** [comment]
@sureshanaparti this is still wip

**Comment by shwstppr** [comment]
@blueorangutan ui

**Comment by utchoang** [comment]
@blueorangutan ui

**Review by utchoang — APPROVED** [review]
LGTM! Tested on QA

File: ui/src/components/view/StoragePoolSelectView.vue [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+<template>
+  <div>
+    <a-input-search
+      class="top-spaced"
+      :placeholder="$t('label.search')"
+      v-model="searchQuery"
```
Comment (davidjumani) [inline ui/src/components/view/StoragePoolSelectView.vue:23]: On change, the page should be reset to zero

File: ui/src/components/view/StoragePoolSelectView.vue [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+    if (this.suitabilityEnabled) {
+      this.columns.splice(1, 0, { slots: { title: 'suitabilityCustomTitle' }, scopedSlots: { customRender: 'suitability' } }
+      )
+    }
+    this.preselectStoragePool()
+    this.fetchStoragePools()
+  },
+  methods: {
+    fetchStoragePools () {
+      if (this.suitabilityEnabled) {
```
Comment (davidjumani) [inline ui/src/components/view/StoragePoolSelectView.vue:175]: Set loading to true / false before / after api call

File: ui/src/components/view/StoragePoolSelectView.vue (hunk 1 above)
Comment (shwstppr) [reply ui/src/components/view/StoragePoolSelectView.vue:23]: Done, on query change page is set to `1`

File: ui/src/components/view/StoragePoolSelectView.vue (hunk 2 above)
Comment (shwstppr) [reply ui/src/components/view/StoragePoolSelectView.vue:187]: Done

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan package

**Review by borisstoyanov — APPROVED** [review]
LGTM, manually tested

**Comment by NuxRo** [comment]
LGTM!

**Review by NuxRo — APPROVED** [review]

**Comment by weizhouapache** [comment]
@blueorangutan package

**Pull request merged**